app/controllers/api/v1/locations_controller.rb

```diff
@@ -27,9 +27,15 @@ def create
   end
 
   def destroy
-    Location.find(params["id"]).destroy
-    locations = Location.all
-    render json: locations
+    location = Location.find(params["id"])
+
+    if current_user.id == location.user_id
+      Location.find(params["id"]).destroy
```

Comment:
you've already grabbed this location -- replace this with `location.destroy` (DRYer and avoids another db call)